### DISCUSSION TITLE
techn. Fragen zur Funktion

### DISCUSSION THREAD

**Discussion by gjuecat** [wiggal/GEN24_Ladesteuerung#14] [General]
Hallo,

ich hätte eine Grundsatzfrage zum Nutzen der Anwendung:
Soweit ich verstanden habe, kann ich damit die Ladung am Vormittag reduzieren, damit ich (zu Mittag) bei 100% Last des Wechselrichters den Ladevorgang DC-seitig nutzen kann, um noch mehr Energie rauszuholen. D.h. abhängig von Solarprognose wird ermittelt, wann der WR seine 100% AC hat und dann wird Batterie geladen? Wie wird da die Lade-Leistung in die Batterie ermittelt, kommt mir tw. sehr gering vor? 

Ich habe einen GEN24 8.0 (mit dem BYD HVM Speicher) und einen SolarEdge 4K. Wie muss ich nun die Einspeisegrenze im config-File festlegen? Die WR Kapazität habe ich mit 8000 angegeben.

Wäre es möglich und sinnvoll, die Grundlast als aktuelle Last von einem Zähler abzugreifen? Bisher hatte ich 700W Grundlast eingetragen, jetzt ist der Pool in Betrieb und nun habe ich (von 11 bis 17Uhr) 2700W Grundlast. Habe nun die Config angepasst, im Herbst dann wieder retour.

lg
Jürgen

**Comment by wiggal** [comment]
> Hallo,
> 
> ich hätte eine Grundsatzfrage zum Nutzen der Anwendung: Soweit ich verstanden habe, kann ich damit die Ladung am Vormittag reduzieren, damit ich (zu Mittag) bei 100% Last des Wechselrichters den Ladevorgang DC-seitig nutzen kann, um noch mehr Energie rauszuholen. D.h. abhängig von Solarprognose wird ermittelt, wann der WR seine 100% AC hat und dann wird Batterie geladen? Wie wird da die Lade-Leistung in die Batterie ermittelt, kommt mir tw. sehr gering vor?

ja genau, neben der Produktion über einer evtl. vorhandenen Einspeisebegrenzung wird auch die DC-Leistung über der AC-Kapazität in die Batterie geladen, soweit Kapazität vorhanden.
Die Lade-Leistung wird durch verschiedene Faktoren bestimmt und die Größere verwendet:
- Produktion über der Einspeisebegrenzung 70% (Einspeisegrenze in config)
- Produktion über der AC-Leistung (WR_Kapazitaet in config)
- oder Berechnung nach Prognose, damit die Batterie möglichst zum Zeitpunkt "BattVollUm" voll wird.

> 
> Ich habe einen GEN24 8.0 (mit dem BYD HVM Speicher) und einen SolarEdge 4K. Wie muss ich nun die Einspeisegrenze im config-File festlegen? Die WR Kapazität habe ich mit 8000 angegeben.

Einspeisegrenze in config = die vorgegebene Einspeisebegrenzung 70% vom Netzbetreiber. Wenn keine vorhanden am besten größer WR_Kapazitaet setzen.

WR_Kapazitaet in config = ein bisschen unter der wirklichen AC-Leistung des WR bei dir evtl 7800, damit das Programm einen Puffer zum reagieren hat.

Wie der  SolarEdge 4K eingebunden ist, bzw. wie er sich auswirkt kann ich nicht sagen.
Die Einspeisung wird über den GEN24 am Smartmeter abgefragt und die Produktion am GEN24,
ob hier die Werte des SolarEdge 4K jeweils mitgeliefert werden kann ich leider nicht sagen.

> 
> Wäre es möglich und sinnvoll, die Grundlast als aktuelle Last von einem Zähler abzugreifen? Bisher hatte ich 700W Grundlast eingetragen, jetzt ist der Pool in Betrieb und nun habe ich (von 11 bis 17Uhr) 2700W Grundlast. Habe nun die Config angepasst, im Herbst dann wieder retour.

Die Grundlast wird bei der Berechnung von jeder Stundenprognose abgezogen, um den Überschuss zu ermitteln, ein auslesen wird nicht auf die zukünftigen Stunden passen, deshalb eine durchschnittliche Grundlast in der Config.

Ich hoffe deine Fragen sind damit beantwortet,
wenn nicht gerne weitere Nachfragen hier.

Schönen Gruß

**Comment by gjuecat** [comment]
> Wie der SolarEdge 4K eingebunden ist, bzw. wie er sich auswirkt kann ich nicht sagen.
> Die Einspeisung wird über den GEN24 am Smartmeter abgefragt und die Produktion am GEN24,
> ob hier die Werte des SolarEdge 4K jeweils mitgeliefert werden kann ich leider nicht sagen.

Ich habe 2 Fronius Smart Meter, eines auf EVU Seite für Einspeisung/Bezug, eines für den SolarEdge. Die Einspeisung kann bei mir also höher sein als die Produktion vom GEN24.

Ich hätte mal versucht, die Reservierung einzuschalten, da bekomme ich aber dann diese Meldung im Crontab.log

`Reservieungsdatei fehlt, bitte erzeugen oder Reservierung abschalten !!`
Im html-Ordner habe ich eine Datei EV_Reservierung.json liegen. Die /DIR/Watt_Reservierung.json wurde bei mir noch nicht angelegt.

> und Reservierung im Browser aufrufen (IP oder localen Namen des RasberryPi).
> 
> Alle eingetragenen Reservierungen werden in die Datei /DIR/Watt_Reservierung.json geschrieben.
> Ist das Modul eingeschaltet (in /DIR/config.ini -->> PV_Reservierung_steuern = 1) wird die Reservierung beim Laden der Batterie beruecksichtigt.

Wie rufe ich die Reservierung im Browser auf?
Wenn ich die index.php im Browser aufrufe, ist es ziemlich egal, welchen Punkt ich aktiviere und pseichere, ich stelle da keine Änderung im Ladeverhalten fest. Sollte im Feld Prognose (kW) der Wert von der Solarprognose angezeigt werden?

![2023-05-19 16_50_53-PV_Planung](https://github.com/wiggal/GEN24_Ladesteuerung/assets/24219395/6e575acd-101f-490d-89e6-1bd159c66ebb)

**Reply by wiggal** [reply]
> 
> Sollte im Feld Prognose (kW) der Wert von der Solarprognose angezeigt werden?
> 

Ja genau, die Werte der Prognose fehlen, den Pfade zur Prognose- und Resevierungsdatei in html/config.php anpassen.

$PrognoseFile = "/home/GEN24/weatherData.json";
$WattReservierungsFile = "/home/GEN24/Watt_Reservierung.json";

**Comment by gjuecat** [comment]
OK, die Anzeige hier funktioniert nun.
Wenn ich aber auf Hausakkuladung = AUS oder VOLL stelle, hat das keine Auswirkung, es wird kein neuer Wert geschrieben. Ist das so geplant? Dachte ich kann damit die spontane Ladung vom Akku erzwingen?

**Reply by wiggal** [reply]
ja so ist es gedacht, damit kann man z.B. den Akku mit voller Ladung die in der config.ini definiert ist laden.
Die Ladeleistung wird beim nächsten Cronjoblauf der Batteriesteuerung auf den Wechselricher geschrieben.

**Reply by wiggal** [reply]
Reservierungssteuerung muss ein sein
PV_Reservierung_steuern = 1
und die Reservierung muss man nach der Auswahl voll, noch speichern

**Reply by gjuecat** [reply]
Ich bekomme das mit der Reservierung nicht hin. Pfade passen würde ich sagen, aber sobald ich in der Config Reservierung auf 1 stelle, habe ich wieder obige Meldung, dass Reservierungsdatei fehlt.

Es liegt halt nur eine leere Datei im Ordner. Muss da zwingend Inhalt rein? Wie muss die aussehen? Hab da keine Vorlage gefunden.

**Reply by wiggal** [reply]
Sobald du auf "PV-Planung ==>> sichern" klickst, müssen folgende zwei Dateien entstehen.

EV_Reservierung.json direkt im Verzeichnis in dem auch die index.php liegt (hab sie txt umbenannt sonnst kann ich sie nicht hochladen)
[EV_Reservierung.txt](https://github.com/wiggal/GEN24_Ladesteuerung/files/11523906/EV_Reservierung.txt)

und 
/DIR/Watt_Reservierung.json
[Watt_Reservierung.txt](https://github.com/wiggal/GEN24_Ladesteuerung/files/11523911/Watt_Reservierung.txt)

Wenn die bei dir nicht entstehen, dann poste mal die Meldungen vom Webserver, wenn du "PV-Planung ==>> sichern" klickst.
Bei Apache stehen die in /var/log/apache2/error.log

Die Planung sollte in etwar so aussehen:
![grafik](https://github.com/wiggal/GEN24_Ladesteuerung/assets/69506553/8a33659a-8f0c-4894-94e2-f554cfe9ca47)

**Comment by gjuecat** [comment]
Wieder einen Schritt näher, Planung sieht nun so aus und die Fehlermeldung von fehlender Reservierungsdatei ist auch weg. Allerdings springt die Auswahl auf der Seite beim Sichern immer wieder auf AUTO zurück und es wird kein neuer Ladewert geschrieben, ganz egal ob ich AUS, HALB oder VOLL auswähle. Problem liegt in einer Berechtigung lt. Apache Log:

`[Sun May 21 12:14:48.871547 2023] [php7:warn] [pid 151553] [client 192.168.36.10:59874] PHP Warning:  file_put_contents(/home/docker/GEN24Ladesteuerung/Watt_Reservierung.json): failed to open stream: Permission denied in /home/docker/GEN24Ladesteuerung/html/speichern.php on line 23, referer: http://192.168.36.2/index.php`

Welche Rechte muss ich auf die Dateien/Ordner setzen? Soll ich für den gesamten GEN24 Ordner eine eigene Gruppe anlegen (läuft bei mir auf einem Debian Server).

**Reply by wiggal** [reply]
User unter dem der Apache läuft muss Schreibrechte haben.
/etc/apache2/envvars:export APACHE_RUN_USER=...

Entweder Rechte entsprechend ändern, oder wenn sonst nix unter Apache läuft, APACHE_RUN_USER auf user der Dateien setzen

**Comment by mosjka1** [comment]
Hallo Leute,

hab mal eine Verständnisfrage zu folgenden Werten:

```
# Hier die Werten für maximale und minimale Batterieentladung in Prozent
MaxEntladung = 100
EntladungAus = 0
```

Wird damit der SOC Wert gemeint, oder mit welcher Leistung in % die Batterie entladen darf? Bei 50 % also mit 5000 Watt bei einer 10000 Watt Batterie. Was bedeutet die „0“ bei EntladungAus ?


`# Wenn Batterie voll in % dann volle Ladung
BatterieVoll = 97
`
Was bedeutet der Wert 97? Batterie Laden bis max. 97%? Meine lädt aber tatsächlich bis 100%.

Das Ganze betreibe ich an einem Gen24 10.0 + HVS 10.2

Hier meine config. Für evtl. Verbesserungen und Tipps wäre ich dankbar. Danke an dieser Stelle an den Programmierer. Hut ab!

```
# Genauere Erklärungen siehe config.info
[env]
filePathWeatherData = weatherData.json

# Wenn ein zweites Feld mit anderer Ausrichtung vorhanden ist, hier anzahl=2 eintragen
# und die Werte unter [forecast.solar2] eintragen
[pv.strings]
anzahl = 2

[forecast.solar]
lat = 50.XXX
lon = 7.XX
dec = 25
az = -20
kwp = 6.5
dataAgeMaxInMinutes = 10000

[forecast.solar2]
lat = 50.XXX
lon = 7.XXX
dec = 25
az = 75
kwp = 3.5

# Hier die Werte eintragen, wenn die Prognosse mit Solarprognose_WeatherData.py von http://www.solarprognose.de geholt werden soll
[solarprognose]
weatherfile = weatherData.json
accesstoken = g8ruFXThZmu5SBSY-XXXXXXXXXX
item = inverter
id = 3370
type = hourly
dataAgeMaxInMinutes = 100
# Sekunden des sleep = Vorgabe von solarprognose.de
WaitSec = 21
# mögliche Werte mosmix|own-v1|clearsky
algorithm = own-v1
# falls die Anlagengröße in KWp nicht mit der Größe in Solarprognose übereinstimmt (z.B. Erweiterung)
KW_Faktor = 1

[gen24]
# IP Wechselrichter
hostNameOrIp = 192.168.178.223
port = 502

[Ladeberechnung]
# Wenn die Variable "FesteLadeleistung" größer "0" Watt ist, wird der Wert geschrieben.
FesteLadeleistung = 0

# Bis wann (Stunde) soll die Batterie voll sein,
BattVollUm = 17
# Faktor um Batteriekapazitaet fuer spaeter zu sparen. (0 = AUS)
# Frühjahr/Sommerwert
#BatSparFaktor = 1.2
# Herbst/Winterwert
#BatSparFaktor = 0
BatSparFaktor = 1.2
# BatWaitFaktor ist für meine exakt nach Süden ausgerichtete PV-Anlage nötig! (0 = AUS)
# Frühjahr/Sommerwert
#BatWaitFaktor = 4.0
# Herbst/Winter
#BatWaitFaktor = 0.0
BatWaitFaktor = 0.0
BatWaitFaktor_Max = 10

# print_level 1=Ausgabe 0=keine Ausgabe
print_level = 1
# Groester Batterieladewert, der im WR gesetzt werden soll, aber höchstens der maximale mögliche Ladewert!!!
MaxLadung = 3500
# die Einspeisegrenze sollte etwas unterhalb der wirklichen Grenze liegen (1-2%).
Einspeisegrenze = 12000
# AC Kapazität des Wechselrichters um den Überschuss in die Batterie zu laden (etwas darunter bleiben 1-2%)
WR_Kapazitaet = 10000
# Puffer für die Einspeisebegrenzung und Wechselrichterkapazität in Watt,
# da die durch den WR abgeregelte Energiemenge nicht gemessen werden kann.
# muss etwa der WRSchreibGrenze_nachOben entsprechen wenn:
# Einspeisegrenze - wirkliche Einspeisegrenze + Puffer_Einspeisegrenze < WRSchreibGrenze_nachOben, wird die Batterieladung nicht angepasst.
Puffer_Einspeisegrenze = 400
# PV-Leistung reellen Spitzenwert
PV_Leistung_Watt = 11600
Puffer_WR_Kapazitaet = 200

# Grundlast wird verwendet um den aktuellerUeberschuss zu berechnen
Grundlast = 900

# Schaltverzögerung um Schreibzugriffe auf WR zu minimieren
WRSchreibGrenze_nachOben = 450
WRSchreibGrenze_nachUnten = 1700

# Liegt der Batteriestatus unter MindBattLad in % wird voll geladen
MindBattLad = 40
# Wenn Batterie voll in % dann volle Ladung
BatterieVoll = 97
# Wenn Ladung ausgeschaltet den Wert LadungAus in Watt setzen (z.B. 0 oder 10 )
LadungAus = 0

[Reservierung]
# Reservierungsdateiname
PV_ReservieungsDatei = Watt_Reservierung.json
# Hier kann die PV-Reserierung (z.B. für ein E-Auto nachmittags) eingeschatet werden (1=ein, 0=aus)
PV_Reservierung_steuern = 0
# Batterieentladung ein- bzw. ausschalten (1=ein)
Batterieentlandung_steuern = 0
# 3 Faktoren muessen gleichzeitig eintreten, damit die Batterieentladung auf minimum geschaltet wird (siehe auch config.ini.info)
BisLadestandEIN = 80
AbReservierungEIN = 2001
# Hier die Werten für maximale und minimale Batterieentladung in Prozent
MaxEntladung = 100
EntladungAus = 0

[messaging]
# Hier Daten für Pushmessage auf Handy
# Push_Message_EIN (1=EIN/0=AUS)
Push_Message_EIN = 0
Push_Message_Url = ntfy.sh/GEN24_mosjka1

[Fallback]
# Ab hier Konfiguration der Fallbackfunktion
# Werte 0, 1 oder 2
Fallback_on = 2
# Fallbackzeit = (Fallback_Zeitabstand_Std * 3600 + Cronjob_Minutenabstand *60 *0.9)
Fallback_Zeitabstand_Std = 5
Cronjob_Minutenabstand = 5
```

**Reply by wiggal** [reply]
> ```
> # Hier die Werten für maximale und minimale Batterieentladung in Prozent
> MaxEntladung = 100
> EntladungAus = 0
> ```
> 
> Wird damit der SOC Wert gemeint, oder mit welcher Leistung in % die Batterie entladen darf? Bei 50 % also mit 5000 Watt bei einer 10000 Watt Batterie. Was bedeutet die „0“ bei EntladungAus ?

Genau, 
MaxEntladung  ist der Wert in Prozent, mit der die Batterie ENTladen wird, wenn die Entladung EINgeschaltet ist (siehe Bedingungen).
EntladungAus ist der Wert in Prozent, mit der die Batterie ENTladen wird, wenn die Entladung AUSgeschaltet ist.

Da in deiner config.ini die Batterieentladung ausgeschaltet ist (Batterieentlandung_steuern = 0), haben die Werte keine Auswirkung.

> 
> `# Wenn Batterie voll in % dann volle Ladung BatterieVoll = 97 ` Was bedeutet der Wert 97? Batterie Laden bis max. 97%? Meine lädt aber tatsächlich bis 100%.
> 
Die Batterieladesteuerung berechnet Ladewerte, bis der Ladestand bei BatterieVoll in Prozent ist, dann wird die Ladung auf MaxLadung gestellt. Bei einem Batteriestand nahe 100% werten die berechneten Ladewerte sehr instabil und es wird sehr häufig auf den WR geschrieben

Ich hoffe ich habs verständliche erklärt
Deine Konfig schau ich mir später durch
Schönen Gruß
Wiggal

**Reply by wiggal** [reply]
Hab deine Konfig überflogen, scheint soweit OK, 
folgende Kleinigkeiten sind mir aufgefallen:


```
> # Sekunden des sleep = Vorgabe von solarprognose.de
> WaitSec = 21
```

Bei solarprognose.de unter "API Übersicht" wird dir die genaue Minute und Sekunde vorgegeben in der du deine Prognose abrufen sollst:

Beispiel von mir:
Minute für Cronjob: 	8 (bitte fügen Sie ein sleep 21s vor der API-Abfrage ein)

Kann Zufall sein, aber wenn bei dir die Sekunde nicht 21 bzw. die Minute nicht 8 ist, bitte die Sekunden in den config.ini anpassen, und die Minuten im Cronjob. Damit schonen wir die Server von solarprognose.de

```
> # die Einspeisegrenze sollte etwas unterhalb der wirklichen Grenze liegen (1-2%).
> Einspeisegrenze = 12000
```
Vermutlich hast du keine Einspeisebegrenzung, dann OK

```
> # PV-Leistung reellen Spitzenwert
> PV_Leistung_Watt = 11600
> Puffer_WR_Kapazitaet = 200
```

Da du vermutlich keine Einspeisebegrenzung hast, hast du vermutlich genügend Akkukapazität und könntest den Wert "Puffer_WR_Kapazitaet = 400" setzen. Damit hast du weniger Verluste an der AC-Wechselrichtergrenze.


Schönen Gruß
weitere Fragen gerne hier
